### PR TITLE
fix(aws-lambda): mark as non-npm and remove from `attr.json`

### PR DESCRIPTION
**Pull request by G-Rath**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] ~Test the change in your own code. (Compile and run.)~
- [x] ~[Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.~
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/awspilot/cli-lambda-deploy/issues/7#issuecomment-1732368393
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.
---

From my reading this is the correct thing to do - this package provides types for the AWS Lambda _service_ rather than the [`aws-lambda`](https://www.npmjs.com/package/aws-lambda) npm package; so it should not compared to that package and not considered published on NPM.

### CODE REVIEW

File: types/aws-lambda/package.json [hunk 1]
```diff
@@ -2,6 +2,8 @@
     "private": true,
     "name": "@types/aws-lambda",
     "version": "8.10.9999",
+    "nonNpm": true,
```
Comment (jakebailey) [inline types/aws-lambda/package.json:5]: Probably this should be "conflict"?

Comment (jakebailey) [reply types/aws-lambda/package.json:5]: (My thought was that "true" was supposed to fail if there was a package on npm so if it's not failing now that's confusing)

Comment (jakebailey) [reply types/aws-lambda/package.json:5]: No, I was right: https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/7937875943/job/21675882216?pr=68667#step:11:57

**Comment by simonbuchan** [comment]
Correct on theory; but I don't know nothin about the DT machinery here!

File: types/aws-lambda/package.json (hunk 1 above)
Comment (G-Rath) [reply types/aws-lambda/package.json:5]: ok sweet, thanks for the quick response! I couldn't find any documentation on this except for the readme which mentions "`nonNpm` is a thing that could be `true`" 🤷 

I'll update it to `conflict`

Comment (jakebailey) [reply types/aws-lambda/package.json]: Yes, this needs more docs. will make a note of that so I remember.

**Review by jakebailey — APPROVED** [review]

**Comment by G-Rath** [comment]
Ready to merge

**Pull request merged**